### ISSUE TITLE
OxygenOS discussion

### ISSUE THREAD

**Issue by Xalaxis** [urbandroid-team/dont-kill-my-app#30]
Hi all,

As an Urbandroid, Sleep as Android (I also have a Sleep phaser 🙂) and OnePlus user I'd like to note that I've never had any problems with Sleep as Android function, at least since the OnePlus 6.  I believe the problems with OxygenOS's battery interference came from three things:

1. Problems with OxygenOS's 'Deep Optimization' function being turned on by default in the past, which turns normal battery optimization into super crazy block everything 'optimization'.  This feature has gone by more than just this name I believe.  I think this feature is now **disabled by default**
2. Problems with OyxgenOS moving apps back to 'Battery Optimized' automatically without warning (bug?)
3. The 'App Auto-Launch' problem previously mentioned.

As far as I'm aware, all these problems are fixed/resolved on the latest OnePlus firmware.  At the very least I've had no problem with any of the OxygenOS firmware versions on the OnePlus 6.  I'd greatly appreciate other OnePlus users feedback on whether number 1. was actually causing their problems with OnePlus battery optimization.

**Comment by Xalaxis** [comment]
Users discussing how these battery issues seem not to be a problem with OnePlus 6 devices.

http://disq.us/p/1yznlqu

![image](https://user-images.githubusercontent.com/12937964/51542526-5f35f000-1e53-11e9-9453-bf905b16d5d2.png)

**Comment by elich11** [comment]
I'm on 9.0.3 no battery optimizations on the app, and no deep battery optimizations, use the latest sleep beta - 201190123.
oxygenOS closes the app every night, currently no solution/wordaround.

Eli

**Comment by elich11** [comment]
> @elich11 What device are you using?

OP6

**Comment by Xalaxis** [comment]
@elich11 We are running exactly the same config (device, OxygenOS version) except for the fact that I haven't excluded Sleep from battery optimizations and I'm running version 20181212 (the latest stable(?)).

Are you sure that version number is correct, it seems very strange.

**Comment by elich11** [comment]
Yep, its a typo, my version is 20190123.
I will add it to the optimized list and check it (although the optimized list means its more likely to be removed from the background).

Thanks for the info,
Eli

**Comment by Xalaxis** [comment]
@elich11 Yeah, if it turns out that apps excluded from optimization are being killed but those included aren't, then that's a pretty big bug.

**Comment by elich11** [comment]
@Xalaxis after one night test, this workaround works perfectly.
The app doesn't get killed!
Good work.

Eli

**Comment by Xalaxis** [comment]
@elich11 Great!

@Artaud, do you want to try and reproduce?

**Comment by elich11** [comment]
@Xalaxis after more tests, I think that what fixed it is the device is plugged in to the charger all night.

Eli

**Comment by Xalaxis** [comment]
@elich11 Hmm, so perhaps this is somehow a (device) sleep related issue.  I don't think I've seen that problem myself, but I can't risk testing this until Wednesday night.  I'll give it a test then (not having the device plugged in).

**Comment by elich11** [comment]
@Xalaxis second night it works (which is unpresedented for me), I think its a legit workaround.

Eli

**Comment by Xalaxis** [comment]
@elich11 I've just tested with my phone, Sleep was not killed during the night.  Sorry for the delay in getting back to you.  Did you confirm whether the charger seemed to effect survival through the night, or whether it was the previously thought battery-optimization settings?

**Comment by elich11** [comment]
@Xalaxis Yes, I'm using it without and issue when the charger is connected for about 2 weeks now.
I will run a test tonight without the charger to make sure the issue is reproduced.

**Comment by elich11** [comment]
@Xalaxis the issue reproduced without a charger, even when using the latest os version (9.0.4).

**Comment by Xalaxis** [comment]
@elich11 Darn, I'm running the same version (9.0.4), no issues 🙁 (even when not on charge) .  This is so puzzling.

**Comment by eminemoholic** [comment]
It's mid 2019 and I'm still looking for a reliable way to run my foreground location service on a Oneplus device. Any workaround? :(

**Comment by Xalaxis** [comment]
> It's mid 2019 and I'm still looking for a reliable way to run my foreground location service on a Oneplus device. Any workaround? :(

Hi @eminemoholic , we've not been having any issues with *foreground* processes.  What's the symptoms of your problem?

**Comment by eminemoholic** [comment]
> 
> 
> > It's mid 2019 and I'm still looking for a reliable way to run my foreground location service on a Oneplus device. Any workaround? :(
> 
> Hi @eminemoholic , we've not been having any issues with _foreground_ processes. What's the symptoms of your problem
For me the notification attached with the foreground service disappears after sometime and thus the service gets killed, without triggering any of the lifecycle events.

**Comment by Xalaxis** [comment]
@eminemoholic Have you ruled out an extreme out-of-memory state?

**Comment by Xalaxis** [comment]
So now with the release of the Don't kill my app tester, I think we can empirically say that at least my OnePlus 6 seems to not encounter any of these issues

![Screenshot_20210506-215722](https://user-images.githubusercontent.com/12937964/117364820-4af84e80-aeb6-11eb-8be3-580354cce602.jpg)
